### PR TITLE
Add warnings on Comments.vote failures

### PR DESCRIPTION
**Pull request by Lucas-C**
I had an issue with my bloom filter, and those log messages helped me a lot to figure it out.

In the end I simply cleared them all:
```python
import sqlite3
from isso.compat import buffer
from isso.utils import Bloomfilter
bf = buffer(Bloomfilter(iterable=["127.0.0.0"]).array)
with sqlite3.connect('.../isso.db') as con:
    con.execute('UPDATE comments SET voters=?', (bf, ))
```

### CODE REVIEW

**Comment by Lucas-C** [comment]
Now I realize the proper fix was not to clear the Bloom filters but to properly configuring remote address resolution based on the HTTP headers provided by my nginx server: https://github.com/posativ/isso/pull/636

**Comment by Lucas-C** [comment]
Just to clarify my previous comment : I still think adding those warnings would be useful ! :)

Please share your thoughts on this, and tell me if ever I need to make changes to the code

**Comment by Lucas-C** [comment]
Note: build is passing -> https://travis-ci.com/github/Lucas-C/isso/builds/160401072

File: isso/db/comments.py [hunk 1]
```diff
@@ -282,10 +286,12 @@ def vote(self, upvote, id, remote_addr):
 
         likes, dislikes, voters = rv
         if likes + dislikes >= 142:
+            logger.warn('Comments.vote(id=%s): likes + dislikes (%s) >= 142', id, likes + dislikes)
```
[the remark this answers is not in the document]
Comment (Lucas-C) [reply isso/db/comments.py:289]: Alright, moving this to `.debug()`
The arbitrary limit of 142 made this worthwhile to warn on

File: isso/db/comments.py [hunk 2]
```diff
@@ -282,10 +286,12 @@ def vote(self, upvote, id, remote_addr):
...
             return {'likes': likes, 'dislikes': dislikes}
 
         bf = Bloomfilter(bytearray(voters), likes + dislikes)
         if remote_addr in bf:
+            logger.warn('Comments.vote(id=%s): a vote has already been registered for this remote address: %s', id, remote_addr)
```
[the remark this answers is not in the document]
Comment (Lucas-C) [reply isso/db/comments.py:294]: I'd say we log both messages as `debug` but also add a `message` field to the JSON output explaining what happened

**Comment by jelmer** [comment]
On Thu, Apr 16, 2020 at 05:29:31AM -0700, Lucas Cimon wrote:
> @Lucas-C commented on this pull request.
> 
> 
> 
> >              return {'likes': likes, 'dislikes': dislikes}
>  
>          bf = Bloomfilter(bytearray(voters), likes + dislikes)
>          if remote_addr in bf:
> +            logger.warn('Comments.vote(id=%s): a vote has already been registered for this remote address: %s', id, remote_addr)
> 
> I'd say we log both messages as `debug` but also add a `message` field to the JSON output explaining what happened
+1

-- 
Jelmer Vernooĳ <jelmer@jelmer.uk>
PGP Key: https://www.jelmer.uk/D729A457.asc

**Comment by Lucas-C** [comment]
Changes made !

**Comment by Lucas-C** [comment]
Travis build is passing : https://travis-ci.com/github/Lucas-C/isso/builds/160566818

**Comment by Lucas-C** [comment]
Thanks !

**Pull request merged**